build: test angular.io/docs examples against RxJS v7

gkalpak opened the pull request:
Extend PR #41793 to run angular.io and docs-examples tests against RxJS v7 on CI (as a quick way to guard against potential regressions).

mary-poppins commented:
You can preview b18dd9d at https://pr42660-b18dd9d.ngbuilds.io/.

mary-poppins commented:
You can preview 4e4d7ca at https://pr42660-4e4d7ca.ngbuilds.io/.

mary-poppins commented:
You can preview 1188781 at https://pr42660-1188781.ngbuilds.io/.

mary-poppins commented:
You can preview 6793089 at https://pr42660-6793089.ngbuilds.io/.

dario-piotrowicz commented on aio/tools/examples/run-example-e2e.js:
```diff
@@ -378,16 +387,22 @@ function getE2eSpecs(basePath, filter) {
-  const e2eSpecGlob = `${filter ? `*${filter}*` : '*'}/${specFile}`;
-  // clang-format on
+
+  console.log(filter);
```
is this `console.log` here on purpose? :slightly_smiling_face:

gkalpak replied:
Ooops 😇
Removed.

mary-poppins commented:
You can preview cf69034 at https://pr42660-cf69034.ngbuilds.io/.

crisbeto commented on aio/tools/examples/run-example-e2e.js:
```diff
@@ -378,16 +387,20 @@ function getE2eSpecs(basePath, filter) {
 }
 
+function filterToGlob(filter) {
+  return Array.isArray(filter) ? `*{${filter.join(',')}}*` : `*${filter}*`;
```
I think that this might not work if the array only has one element. Not sure if that can actually happen at runtime.

gkalpak replied:
When there is only one item (i.e. if there i only one occurrence of `--filter`/`--exclude`), then `filter` will be a string. `filter` will only be an array is there are multiple values.

crisbeto reviewed (approved):
LGTM, aside from the one comment.

gkalpak committed:
test(docs-infra): allow excluding certain docs examples from tests

This commit adds support for excluding certain docs examples from the
command used to run tests. This is useful to run extra tests on CI that
might not be compatible with all examples (for example, run tests with
different versions of a dependency).

In a subsequent commit, this will be used to run tests against RxJS v7
as a quick way to catch potential regressions.

gkalpak committed:
refactor(docs-infra): make angular.io and docs examples compatible with RxJS v7

This commit applies the necessary changes to make angular.io and (most)
docs examples compatible with both RxJS v6 and v7. It also adds new
steps to relevant CI jobs to test angular.io and docs examples against
RxJS v7 as well (to catch potential regressions).

NOTE:
Some of the docs examples are not compatible with RxJS v7 and are
therefore excluded from the tests. The SystemJS-based ngUpgrade
examples in particular are excluded, since they require a different
SystemJS configs to run against RxJS v6 and v7 and the extra complexity
of managing that is not worth it for the 4 affected examples.

gkalpak committed:
fixup! test(docs-infra): allow excluding certain docs examples from tests

mary-poppins commented:
You can preview abb1cd0 at https://pr42660-abb1cd0.ngbuilds.io/.

gkalpak commented:
Marking this for merging.
Thx, @dario-piotrowicz and @crisbeto, for the reviews ✨
